src/candid.ts

```diff
@@ -0,0 +1,233 @@
+      describe: 'Candid Premier API key; get from account management at https://dashboard.candid.org/',
+      demandOption: true,
+      type: 'string',
+    })
+    .option('output-file', {
+      alias: 'write',
+      describe: 'Write organization information to the specified JSON file',
+      normalize: true,
+      type: 'string',
+    })
+    .positional('ein', {
+      describe: 'US tax ID of organization to look up',
+      demandOption: true,
+      type: 'string',
+    })
+    .check(({ ein }) => isValidEin(ein))
+  ),
+  handler: async (args) => {
+    const result = await getCandidProfile(args['candid-api-key'], args.ein);
+
+    if (args.outputFile) {
+      await writeFile(
+        args.outputFile,
+        JSON.stringify(result, null, 2),
+      );
+      logger.info(`Wrote Candid data for ${args.ein} to ${args.outputFile}`);
+    } else {
+      logger.info({ result }, 'Candid result');
+    }
+  },
+};
+
+const updateCommand: CommandModule = {
+  command: 'update <ein>',
+  describe: 'Fetch information about an organization by its EIN, and upload to the PDC',
+  builder: (y) => (y
+    .option('candid-api-key', {
+      describe: 'Candid Premier API key; get from account management at https://dashboard.candid.org/',
+      demandOption: true,
+      type: 'string',
+    })
+    .options(oidcOptions)
+    .option('pdc-api-base-url', {
+      describe: 'Location of PDC API',
+      demandOption: true,
+      type: 'string',
+    })
+    .positional('ein', {
+      describe: 'US tax ID of organization to look up',
+      demandOption: true,
+      type: 'string',
+    })
+    .check(({ ein }) => isValidEin(ein))
+  ),
+  handler: async (args) => {
+    const token = await getToken(
+      args.oidcBaseUrl as string,
+      args.oidcClientId as string,
+      args.oidcClientSecret as string,
+    );
+    const data = await getCandidProfile(
+      args.candidApiKey as string,
+      args.ein as string,
+    );
+    await postPlatformProviderData(
+      args.pdcApiBaseUrl as string,
+      token,
+      args.ein as string,
+      'candid',
+      data,
+    );
+    logger.info(`Wrote data for ${args.ein as string} to PDC`);
+  },
+};
+
+const getEinBaseFieldId = async (
+  baseUrl: string,
+  token: AccessTokenSet,
+) => {
+  const baseFields = await getBaseFields(baseUrl, token);
+  const einBaseField = baseFields.find(({ shortCode }) => (
+    shortCode === 'organization_tax_id'
+  ));
+  if (einBaseField === undefined) {
+    throw new Error('Could not find base field with short code `organization_tax_id`');
+  }
+  return einBaseField.id;
+};
+
+const getEinsFromPdc = async (baseUrl: string, token: AccessTokenSet) => {
+  const einBaseFieldId = await getEinBaseFieldId(baseUrl, token);
+  const proposals = await getProposals(baseUrl, token);
+
+  const eins = new Set(proposals.entries
+    .flatMap((proposal) => proposal.versions[0]?.fieldValues)
+    .filter(<T>(x: T | undefined): x is T => typeof x !== 'undefined')
```

Comment:
My eyes had to pause here for a bit but then it makes sense. The next filter will get no "undefined" fields.

Reply:
Yep, exactly! I believe that should have already been the case, but type-theoretically that wasn't previously guaranteed, and now it's certain.